deepchem/models/torch_models/layers.py

```diff
@@ -3058,6 +3058,135 @@ def forward(self, inputs: torch.Tensor):
+    hidden_layer and it hold results of the convolution while first two are unchanged
+    input tensors.
+
+    Example
+    -------
+    See: MolGANMultiConvolutionLayer for using in layers.
+
+    >>> import torch
+    >>> import torch.nn as nn
+    >>> import torch.nn.functional as F
+    >>> vertices = 9
+    >>> nodes = 5
+    >>> edges = 5
+    >>> units = 128
+
+    >>> layer1 = MolGANConvolutionLayer(units=units, edges=edges, nodes=nodes, name='layer1')
+    >>> adjacency_tensor = torch.randn((1, vertices, vertices, edges))  # added batch dimension
+    >>> node_tensor = torch.randn((1, vertices, nodes))  # added batch dimension
+    >>> output = layer1([adjacency_tensor, node_tensor])
+
+    References
+    ----------
+    .. [1] Nicola De Cao et al. "MolGAN: An implicit generative model
+        for small molecular graphs", https://arxiv.org/abs/1805.11973
+    """
+
+    def __init__(self,
+                 units,
+                 nodes,
+                 activation=F.tanh,
+                 dropout_rate=0.0,
+                 edges=5,
+                 name="",
+                 **kwargs):
+        """
+        Initialize this layer.
+
+        Parameters
+        ---------
+        units: int
+            Dimesion of dense layers used for convolution
+        activation: function, optional (default=Tanh)
+            activation function used across model, default is Tanh
+        dropout_rate: float, optional (default=0.0)
+            Dropout rate used by dropout layer
+        edges: int, optional (default=5)
+            How many dense layers to use in convolution.
+            Typically equal to number of bond types used in the model.
+        name: string, optional (default="")
+            Name of the layer
+        """
+        super(MolGANConvolutionLayer, self).__init__()
+
+        self.activation = activation
+        self.dropout_rate = dropout_rate
+        self.units = units
+        self.edges = edges
+        self.name = name
+
+        self.dense1 = nn.ModuleList(
+            [nn.Linear(nodes, self.units) for _ in range(edges - 1)])
+        self.dense2 = nn.Linear(nodes, self.units)
+        self.dropout = nn.Dropout(self.dropout_rate)
+
+    def forward(self, inputs):
+        """
+        Invoke this layer
+
+        Parameters
+        ----------
+        inputs: list
+            List of two input matrices, adjacency tensor and node features tensors
+            in one-hot encoding format.
+        training: bool
+            Should this layer be run in training mode.
+            Typically decided by main model, influences things like dropout.
+
+        Returns
+        --------
+        tuple(torch.Tensor,torch.Tensor,torch.Tensor)
+            First and second are original input tensors
+            Third is the result of convolution
+        """
+        ic = len(inputs)
+        assert ic > 1, "MolGANConvolutionLayer requires at least two inputs: [adjacency_tensor, node_features_tensor]"
```

Comment:
Instead of an assertion, raise a `ValueError`. Raising errors is generally better than assertions

Reply:
made the requested change